lib/pf/factory/condition/vlanfilter.pm

```diff
@@ -0,0 +1,107 @@
+package pf::factory::condition::vlanfilter;
+
+=head1 NAME
+
+pf::factory::condition::vlanfilter
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::factory::condition::vlanfilter
+
+=cut
+
+use strict;
+use warnings;
+use Module::Pluggable search_path => 'pf::condition', sub_name => '_modules', require => 1;
+
+our @MODULES;
+
+sub factory_for {'pf::condition'}
+
+our %VLAN_FILTER_TYPE_TO_CONDITION_TYPE = (
+    'is'        => 'pf::condition::equals',
+    'is_not'    => 'pf::condition::not_equals',
+    'match'     => 'pf::condition::matches',
+    'match_not' => 'pf::condition::not_matches',
+);
+
+our %VLAN_FILTER_KEY_TYPES = (
```

Comment:
This is for types that are hashes not simple values.
{
node_info => {...},
ssid => 'BigBob',
connection_type => 'pigeon'
}